core/string_api.php

```diff
@@ -445,13 +448,15 @@ function( $p_array ) use( $p_detail_info, $p_fqdn ) {
 		} else {
 			$s_bugnote_link_callback[$p_include_anchor][$p_detail_info][$p_fqdn] =
 				function( $p_array ) {
-					$t_bug_id = bugnote_get_field( (int)$p_array[2], 'bug_id' );
-					if( $t_bug_id && bug_exists( $t_bug_id ) ) {
-						return $p_array[1] .
-							string_get_bugnote_view_url_with_fqdn( $t_bug_id, (int)$p_array[2] );
-					} else {
-						return $p_array[0];
+					$c_bugnote_id = (int)$p_array[2];
+					if( bugnote_exists( $c_bugnote_id ) ) {
+						$t_bug_id = bugnote_get_field( $c_bugnote_id, 'bug_id' );
+						if( $t_bug_id && bug_exists( $t_bug_id ) ) {
+							return $p_array[1] .
+								string_get_bugnote_view_url_with_fqdn( $t_bug_id, $c_bugnote_id );
+						}
 					}
+					return $p_array[0];
```

Comment:
ok, tested this case. The construction is functionally equivalent.
by doing:
`$t_bug_id = bugnote_get_field( $c_bugnote_id, 'bug_id' );`
and then 
`if( $t_bug_id ... `
it's implicitly checking the bugnote exists.
I added the explicit check to make it clearer